### PR TITLE
MM-40840 Server side of export functionality.

### PR DESCRIPTION
**Pull request by crspeller**
#### Summary
Server side of Export functionality. 

Decided to go with this approach for ease of use and to allow for checks to verify that export is being considered when fields are being added.

For an explanation of use see the comment above the Playbooks struct. (Improved wording of that comment welcomed!) 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-40840

### CODE REVIEW

**Comment by lieut-data** [comment]
Swapping myself out for @iomodo.

File: webapp/package-lock.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
         "chrono-node": "2.3.5",
         "core-js": "3.20.2",
         "debounce": "1.2.1",
-        "luxon": "2.3.0",
+        "luxon": "2.0.2",
```
Comment (iomodo) [inline webapp/package-lock.json:13]: why do we have this change?

File: server/app/export.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+	return exported
+}
+
+// GeneratePlaybookExport returns a playbook in export format.
+// Fields marked with the stuct tag "export" are included using the given string.
+func GeneratePlaybookExport(playbook Playbook) ([]byte, error) {
+	export := getFieldsForExport(playbook)
+	export["version"] = CurrentPlaybookExportVersion
+	export["checklists"] = generateChecklistExport(playbook.Checklists)
```
Comment (iomodo) [inline server/app/export.go:53]: why not do the recursive implementation, it would be resilient to all the struct changes in the future.

**Review by iomodo — APPROVED** [review]
Looks great @crspeller 
Just couple of questions below

File: webapp/package-lock.json (hunk 1 above)
Comment (crspeller) [reply webapp/package-lock.json:13]: That's the package-lock. I am guessing when that other PR went in it wasn't updated so this PR is doing it.

File: server/app/export.go (hunk 2 above)
Comment (crspeller) [reply server/app/export.go:53]: We could possibly detect slice type and do that. But I think I would rather force some thought here. If we create some super complicated structure we can change it.

**Review by calebroseland — APPROVED** [review]

**Pull request merged**